core/Enclave/enclave_hsm.edl

```diff
@@ -67,49 +67,42 @@ enclave {
-                [in, size=context_len] const uint8_t *context, size_t context_len,
-                [out, size=plain_key_len] uint8_t *plain_key, size_t plain_key_len,
-                [out, size=encrypted_key_len] uint8_t *encrypted_key, size_t encrypted_key_len);
-
-        public sgx_status_t enclave_export_datakey(uint32_t cmk_spec,
-                [in, size=cmk_blob_size] const uint8_t *cmk_blob, size_t cmk_blob_size,
-                [in, size=context_len] const uint8_t *context, size_t context_len,
-                [in, size=encrypted_key_len] uint8_t *encrypted_key, size_t encrypted_key_len,
-				uint32_t uk_spec,
-				[in, size=uk_blob_size] const uint8_t *uk_blob, size_t uk_blob_size,
-                [out, size=new_encrypted_key_len] uint8_t *new_encrypted_key, size_t new_encrypted_key_len);
-
-        public sgx_status_t enclave_create_rsa_key([out, size=cmk_blob_size] uint8_t *cmk_blob, uint32_t cmk_blob_size,
-                                               [out] uint32_t *req_blob_size);
-        public sgx_status_t enclave_rsa_sign([in, size=cmk_blob_size] const uint8_t *cmk_blob, size_t cmk_blob_size,
-                                         [in, size=data_len] const uint8_t *data, uint32_t data_len,
-                                         [out, size=signature_len] uint8_t *signature, uint32_t signature_len);
-        public sgx_status_t enclave_rsa_verify([in, size=cmk_blob_size] const uint8_t *cmk_blob, size_t cmk_blob_size,
-                                           [in, size=data_len] const uint8_t *data, uint32_t data_len,
-                                           [in, size=signature_len] const uint8_t *signature, uint32_t signature_len,
-                                           [out] bool* result);
-        public sgx_status_t enclave_rsa_encrypt([in, size=cmk_blob_size] const uint8_t *cmk_blob, size_t cmk_blob_size,
-                                            [in, size=plaintext_len] const uint8_t *plaintext, uint32_t plaintext_len,
-                                            [out, size=ciphertext_len] uint8_t *ciphertext, uint32_t ciphertext_len);
-        public sgx_status_t enclave_rsa_decrypt([in, size=cmk_blob_size] const uint8_t *cmk_blob, size_t cmk_blob_size,
-                                            [in, size=ciphertext_len] const uint8_t *ciphertext, uint32_t ciphertext_len,
-                                            [out, size=plaintext_len] uint8_t *plaintext, uint32_t plaintext_len,
-                                            [out] uint32_t *req_plaintext_len);
+        public sgx_status_t enclave_create_aes_key([out, size=cmk_blob_size] uint8_t *cmk_blob, uint32_t cmk_blob_size);
+        
+        public sgx_status_t enclave_create_rsa_key([out, size=cmk_blob_size] uint8_t *cmk_blob, uint32_t cmk_blob_size);
+
+        //public sgx_status_t enclave_create_ec_key(/* param */);
+
+        //public sgx_status_t enclave_create_hmac_key(/* param */);
```

Comment:
you need to be notice that, the number of ecall is as smaller as good.

so you better to combine the similar function into one. such as:
createkey()
encrypt()
decrypt()
sign()
verify()
....